diff --git a/lib/app/configuration/default-hooks.js b/lib/app/configuration/default-hooks.js
index 6454506..b0f590e 100644
--- a/lib/app/configuration/default-hooks.js
+++ b/lib/app/configuration/default-hooks.js
@@ -9,6 +9,7 @@
 
 module.exports = {
   'moduleloader': true,
+  'events-to-elasticsearch': true,
   'logger': true,
   'request': true,
   'orm': 'sails-hook-orm',
diff --git a/lib/hooks/events-to-elasticsearch/index.js b/lib/hooks/events-to-elasticsearch/index.js
new file mode 100644
index 0000000..22852ba
--- /dev/null
+++ b/lib/hooks/events-to-elasticsearch/index.js
@@ -0,0 +1,106 @@
+/**
+ * Module dependencies.
+ */
+
+var request = require("request");
+
+var events = [
+  'elasticsearch-listeners-added',
+  'hook:blueprints:loaded',
+  'hook:controllers:loaded',
+  'hook:cors:loaded',
+  'hook:csrf:loaded',
+  'hook:enforceRestfulSesssions:checked',
+  'hook:events-to-elasticsearch:loaded',
+  'hook:grunt:done',
+  'hook:grunt:error',
+  'hook:grunt:loaded',
+  'hook:http:listening',
+  'hook:http:loaded',
+  'hook:i18n:loaded',
+  'hook:index.js:loaded',
+  'hook:logger:loaded',
+  'hook:moduleloader:loaded',
+  'hook:policies:bound',
+  'hook:policies:loaded',
+  'hook:pubsub:loaded',
+  'hook:pubsub:reloaded',
+  'hook:request:loaded',
+  'hook:responses:loaded',
+  'hook:services:loaded',
+  'hook:session:loaded',
+  'hook:userconfig:loaded',
+  'hook:userhooks:loaded',
+  'hook:views:loaded',
+  'hooks:builtIn:ready',
+  'lifted',
+  'lower',
+  'middleware:registered',
+  'ready',
+  'route:typeUnknown',
+  'router:after',
+  'router:before',
+  'router:bind',
+  'router:request:404',
+  'router:request:500',
+  'router:reset',
+  'router:route',
+  'router:unbind',
+];
+
+function handleRequestResponse(error) {
+  if (error) {
+    console.log("elasticsearch error", error);
+  }
+}
+
+function postToES(sails, event) {
+  return function(data) {
+    sails.log.debug("Adding " + event + " to elasticsearch");
+    request({
+      url: "http://localhost:9200/sails/events",
+      method: 'POST',
+      json: true,
+      body: {
+        event: event,
+        at: Date.now(),
+        data: data || '',
+      }
+    }, handleRequestResponse);
+  }
+}
+
+module.exports = function(sails) {
+
+
+  /**
+   * Expose `logger` hook definition
+   */
+
+  return {
+
+
+    defaults: {
+    },
+
+
+    configure: function() {
+    },
+
+
+    /**
+     * Initialize is fired when the hook is loaded,
+     * but after waiting for user config.
+     */
+
+    initialize: function(cb) {
+      events.forEach(function(e) {
+        sails.on(e, postToES(sails, e));
+      });
+
+      sails.emit('elasticsearch-listeners-added');
+      return cb();
+    }
+
+  };
+};
diff --git a/package.json b/package.json
index 120e7d6..cce1fd7 100644
--- a/package.json
+++ b/package.json
@@ -82,6 +82,7 @@
     "prompt": "0.2.14",
     "rc": "1.0.1",
     "reportback": "~0.1.9",
+    "request": "^2.68.0",
     "rttc": "9.3.3",
     "sails-disk": "~0.10.9",
     "sails-generate": "~0.13.0",
